dev/source/docs/code-overview-adding-a-new-parameter.rst

```diff
@@ -264,10 +264,41 @@ From time to time parameters need to be altered or renamed. ArduPilot has capabi
 
     * Set the AP_PARAM_KEY_DUMP definition to "1" [here in AP_Param.h](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Param/AP_Param.h#L36)
 
-    * Change the delay here from 1ms to 2ms [here in AP_Param::show_all](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Param/AP_Param.cpp#L2414)
-
-    * Remove the #if / #endif from [here in Copter's system.cpp](https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/system.cpp#L262)
-
     * Start the old code in SITL and all the parameter names and their magic numbers will be displayed
 
     * Copy-paste from Copter's Parameters.cpp file's [existing parameter conversion tables like the ones done for the attitude controller's filters](https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/Parameters.cpp#L1219).
+
+.. tip::
+
+    If you are using xterm to open the debug terminal you may find that the list of parameters is too long for the terminal's history to find the parameter(s) you are looking for.  To resolve this, create a file called *.Xresources* in your 
+    home directory.  Using your preferred text editor add the following lines and save:
+
+    ::
+
+        XTerm.vt100.saveLines: 4096
+        Term.vt100.scrollBar: true
+
+    Then, add ``xrdb ~/.Xresources`` to the end of your *.bashrc* and save.  Finally, re-load your *.bashrc* file with the command ``source ~/.bashrc``.  Upon re-running SITL, the xterm window should now have a scroll bar and a longer history allowing you to 
+    find the parameters you need.
+
+Parameter Conversion Expiration
+===============================
+
+Parameter conversions are necessary to ensure continuity for users as they upgrade to newer versions of ArduPilot.  However, the conversion process does utilise some resources on boot and does add to the flash cost of the code. 
+Therefore we don't keep the conversion code forever more.  The agreed process is to remove the conversion when both of the following conditions are true:
+
+    * The stable version of the code has incremented by two major versions since the conversion was added.
+    * The conversion was implemented more than 2 years ago.
+
+For example:  A conversion was added to master on the 14-Aug-2021, at which point the latest version of the stable release is 4.0.7. Therefore, the conversion can be removed after 14-Aug-2023 and once the stable release is greater than 4.2.
+
+To make it easier for to find and remove the conversions, please add the following as a comment to any parameter conversion that you add:
+
+::
+
+    /*
+    PARAMETER_CONVERSION
+    Remove when: stable release > 4.2 && date > 14-Aug-2023
```

Comment:
discussed on the dev call, and decided that adding the date the conversion code was added rather than the date to remove. The date to remove depends on how widely used a version is.

Reply:
Done.  It is simply search string and date added now.